extension/src/time_weighted_average.rs

```diff
@@ -51,7 +51,19 @@ impl<'input> TimeWeightSummary<'input> {
         prev: Option<TimeWeightSummary>,
         next: Option<TimeWeightSummary>,
     ) -> TimeWeightSummary<'static> {
-        assert!(interval_start <= self.first.ts && interval_start + interval_len > self.last.ts);
+        assert!(
+            interval_start <= self.first.ts,
+            "Interval start ({}) must be before first timestamp ({})",
```

Comment:
what do you think about 
```suggestion
            "Interval start ({}) must be at or before first timestamp ({})",
```
since we're doing a `<=` comparison?